probe_src/probe_py/cli.py

```diff
@@ -116,4 +116,98 @@ def dump(
                     print(op.data)
                 print()
 
+@app.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    ),
+)
+def ssh(
+        ssh_args: list[str],
+        debug: bool = typer.Option(default=False, help="Run verbose & debug build of libprobe"),
+):
+    """
+    Wrap SSH and record provenance of the remote command.
+    """
+
+    one_arg_options = set("BbcDEeFIiJLlmOoPpRSWw")
+
```

Comment:
Also define the `no_arg_options` and parse options that appear after the destination if they begin with a dash or follow a one-arg option.

Reply:
UNTIL we reach a `CMD`, which is an argument that cannot be parsed as an argument.

Reply:
```
ssh remote -t ls
ssh remote ls -t
```

In the first example `-t` is an option of `ssh`.

In the second example `-t` is an option of `ls`. We can tell the difference because `ls` is a command that cannot be interpreted as an option, so everything after it must be an option of `ls` not `ssh`.

Reply:
This is parsable as a finite state automaton.

States := {not seen destination, not seen destination & last arg was a one-arg option, seen destination, seen destination & last arg was a one-arg option, seen command}. You can have one state variable holding a string or you can have three boolean variables (seen destination, last arg was one-arg-option, seen command) whose combination represents the state.

Alphabet := {one-arg option, zero-arg option, not-an-option}

Reply:
Don't go overboard and use a finite state automaton class or transition matrix; FSAs can be implemented nicely with just a for-loop, state variable(s), and an if-elif-...-elif-else.